src/tui/mod.rs

```diff
@@ -0,0 +1,144 @@
+mod terminal;
+
+use crate::async_buffer_redirect::AsyncBufferRedirect;
+use crate::ShutdownHandle;
+use ansi_to_tui::IntoText;
+use crossterm::event::Event;
+use crossterm::event::EventStream;
+use crossterm::event::KeyCode;
+use crossterm::event::KeyModifiers;
+use crossterm::event::MouseEventKind;
+use miette::miette;
+use miette::IntoDiagnostic;
+use miette::WrapErr;
+use ratatui::prelude::Buffer;
+use ratatui::prelude::Rect;
+use ratatui::widgets::Paragraph;
+use ratatui::widgets::Widget;
+use ratatui::widgets::Wrap;
+use std::cmp::min;
+use tokio::io::AsyncReadExt;
+use tokio::io::DuplexStream;
+use tokio_stream::StreamExt;
+
+#[derive(Default)]
+struct Tui {
+    quit: bool,
+    scrollback: Vec<u8>,
+    // TODO(evan): Follow output when scrolled to bottom
+    scroll_offset: usize,
```

Comment:
When the user is scrolled to the bottom of the scrollback (the default if they haven't scrolled at all), new output should push old output up and off the screen. Like a "follow logs" option in a CI service.